### ISSUE TITLE
Linting on brick imports in VSCode

### ISSUE THREAD

**Issue by ttamg** [DavidVujic/python-polylith#49]
**Is your feature request related to a problem? Please describe.**
I'm loving exploring Polylith and the poetry tool and I think it has legs to be very helpful in the sort of systems I am building.  

One small niggle when developing in VSCode is that the imports have linting errors in some circumstances.

The issue is that in this framework we do imports from the python package rather than from the directory structure in the brick structure.  For example this is a directory structure where my workspace namespace is `fruity`

```
- bases
  - fruity 
    - grapefruit
- components
  - fruity
    - banana
    - pear
```

Let's say I want to use the `banana` component in my `grapefruit` base, the Polylith import I need to use is from the workspace namespace ... 

```python
# bases/grapefruit/core.py
from fruity.banana import banana
...
```

At the start of a project this generates a linting error as the `fruity.banana` path is not found in the directory structure.  This makes sense as this import is not going to be a **from a directory** import but is going to be a **from the built poetry package** import.

When I made my correct `packages` imports in the master root `pyproject.toml` and did a `poetry build` it built the package and installed it.  Now the linting issues disappear as the package is now installed.  

**Describe the solution you'd like**

There are three issues here I think:

1) I could see this was going to happen but some people may be confused or do the absolute imports which would be incorrect.  I think this could be fixed with some examples and documentation.  I am happy to contribute to this if that is useful.

2) If I run tests or similar now, these imports will be be using what was last built in `poetry build`.  That might be out of date with my current code and cause confusion.  Doing a `poetry build` as part of your pre-run and pre-test activities will be important but may be easy to forget

3) If I want to import `banana` into `pear`, both of which are components.  The correct import does in fact work.  What is now unclear to me when I run my code is whether the `poetry` banana is being used or the directory `banana` version.  They should be the same of course.

I think the most important thing is to find a way to include the `poetry build` into the workflow so that it stays up to date.  I don't know how best to do this without getting in the way of the development workflow but wanted to raise the issue as there may be better ideas. 

**Describe alternatives you've considered**
I also wondered if some sort of VSCode some sort of plugin could use the root `pyproject.toml` `packages` to suppress linting errors, but I think this is not the real issue.  The real issue is that you need to have done a `build`.

H

**Comment by DavidVujic** [comment]
Thank you for sharing feedback, much appreciated! ⭐ 

I'll try this out and look at your feedback more carefully. 

Meanwhile, from the top of my head: what I think would be enough is to run `poetry install` only once, that will create a virtual environment. In there, it should be a `<your-repo-name>.pth` file with paths that should make the `bases` and `components` recognizable for the editor/IDE. I'll dig deeper into this, but think that it is the virtual environment that is needed here.

**Comment by ttamg** [comment]
Thinking about this a bit more.  This is of course because Polylith is set up for a compiled language paradigm.  Much of the Python thinking (and my head) is in the 'compiles on the fly' mode.  And the `poetry build` to create a wheel represents a compilation type of stage.

So I guess I may be saying that mypy is the way to go as you had in your examples @DavidVujic .  I haven't used it so far.  But does that resolve the issue?

**Comment by DavidVujic** [comment]
I think it should work on-the-fly - if it doesn't I probably have created a 🐛 somewhere 😄  

Once a virtual environment is there, and if the `pyproject.toml` has the `packages` section set with one or more components/bases pointed to the components/bases. `Poetry` should then generate that `pth` file correctly and the editor should not warn about any missing packages or such. This should only be done once, since all code added is within a `components/namespace` structure.

**Comment by DavidVujic** [comment]
Aha, now I understand what you meant with `mypy`!

Yes, if you use that in your repo, the configuration need to be there for mypy to understand how the code is structured (namespaces).

**Comment by DavidVujic** [comment]
I also realize now that I should put this documentation about structure and Mypy here, and not only in the examples repo. Sorry! https://github.com/DavidVujic/python-polylith-example#the-loose-theme

**Comment by ttamg** [comment]
> I also realize now that I should put this documentation about structure and Mypy here, and not only in the examples repo. Sorry! https://github.com/DavidVujic/python-polylith-example#the-loose-theme

I did set up with the loose theme as that seemed logic.  But haven't set up Mypy at all.  Something I will have to get my head around!

**Comment by DavidVujic** [comment]
I think the issues described here are solved now. There's also a brand new [Documentation website](https://davidvujic.github.io/python-polylith-docs/) for this project!

Apologies if I have missed out on anything here. If so, please create a new issue 🙏
